metricbeat/module/aerospike/aerospike_test.go

```diff
@@ -96,3 +99,103 @@ func TestParseInfo(t *testing.T) {
 		assert.Equal(t, test.expected, result, test.Name)
 	}
 }
+
+func getStringPointer(value string) *string {
+	return &value
+}
+
+func getBoolPointer(value bool) *bool {
+	return &value
+}
+
+func TestParseClientPolicy(t *testing.T) {
+	sampleUser := "Test"
+	samplePassword := "MySecretPassword"
+	sampleClusterName := "TestCluster"
+
+	UserPasswordClientPolicy := as.NewClientPolicy()
+	UserPasswordClientPolicy.User = sampleUser
+	UserPasswordClientPolicy.Password = samplePassword
+
+	TLSPolicy := as.NewClientPolicy()
+	tlsconfig, _ := tlscommon.LoadTLSConfig(&tlscommon.Config{Enabled: getBoolPointer(true)})
+	TLSPolicy.TlsConfig = tlsconfig.ToConfig()
+
+	ClusterNamePolicy := as.NewClientPolicy()
+	ClusterNamePolicy.ClusterName = sampleClusterName
+
+	tests := []struct {
+		Name                 string
+		Config               Config
+		expectedClientPolicy *as.ClientPolicy
+		expectedErr          error
+	}{
+		{
+			Name:                 "No configurations lead to default policy",
+			Config:               Config{},
+			expectedClientPolicy: as.NewClientPolicy(),
+			expectedErr:          nil,
+		},
+		{
+			Name: "Username and password are honored",
+			Config: Config{
+				User:     getStringPointer(sampleUser),
+				Password: getStringPointer(samplePassword),
+			},
+			expectedClientPolicy: UserPasswordClientPolicy,
+			expectedErr:          nil,
+		},
+		{
+			Name: "Username is set but Password is not",
+			Config: Config{
+				User: getStringPointer(sampleUser),
+			},
+			expectedClientPolicy: UserPasswordClientPolicy,
+			expectedErr:          fmt.Errorf("if username is set, password should be set too"),
+		},
+		{
+			Name: "Password is set but Username is not",
+			Config: Config{
+				Password: getStringPointer(samplePassword),
+			},
+			expectedClientPolicy: UserPasswordClientPolicy,
+			expectedErr:          fmt.Errorf("if password is set, username should be set too"),
+		},
+		{
+			Name: "TLS Declaration",
+			Config: Config{
+				TLS: &tlscommon.Config{
+					Enabled: getBoolPointer(true),
+				},
+			},
+			expectedClientPolicy: TLSPolicy,
+			expectedErr:          nil,
+		},
+		{
+			Name: "Cluster Name Setting",
+			Config: Config{
+				ClusterName: getStringPointer(sampleClusterName),
+			},
+			expectedClientPolicy: ClusterNamePolicy,
+			expectedErr:          nil,
+		},
+	}
+
+	for _, test := range tests {
+		result, err := ParseClientPolicy(test.Config)
+		if err != nil {
+			if test.expectedErr != nil {
+				assert.Equal(t, test.expectedErr.Error(), err.Error())
+				continue
+			}
+			t.Error(err)
+			continue
+		}
+		assert.Equal(t, test.expectedClientPolicy.User, result.User, test.Name)
+		assert.Equal(t, test.expectedClientPolicy.Password, result.Password, test.Name)
+		assert.Equal(t, test.expectedClientPolicy.ClusterName, result.ClusterName, test.Name)
```

Comment:
would you mind switching to `Equalf` and providing a description of what didn't match?  

Right now if this fails in CI we will get an error like `"A" does not equal "B"`.  It is a lot easier to debug if we get something like `Aerospike policy username is wrong. got: "A" expected: "B"`

Reply:
Hi,
Thank you for the feedback. I don't mind at all ^^. So something like:

```go
assert.Equalf(t, test.expectedClientPolicy.User, result.User, "Aerospike policy [Test '%s'] username is wrong: expected '%s' got '%s'", test.Name, test.expectedClientPolicy.User, result.User)
```

Or something like:
```go
assert.Equalf(t, test.expectedClientPolicy.User, result.User, "Aerospike policy [Test '%s'] username is wrong", test.Name, test.expectedClientPolicy.User, result.User)
```
??

Should I also modify the existing tests to use Equalf?

Reply:
I don't think you need the test name, when it errors out go test will give us that.  I prefer the ones that have expected and got (or something like that).

One thing I think that helps is to fail the test and just double check that the error message gives you enough context that you have some idea of where the problem is.  For example telling me that the Username is what didn't match, the value we got, and the expected value goes a long way to helping understand where the problem might be.

I think you can just focus on the tests you are adding.  That helps limit the scope of the PR.